pkg/osquery/tables/profiles/profiles.go

```diff
@@ -0,0 +1,163 @@
+//+build darwin
+
+// Package profiles provides a table wrapper around the various
+// profiles options.
+//
+// As the returned data is a complex nested plist, this uses the
+// dataflatten tooling. (See
+// https://godoc.org/github.com/kolide/launcher/pkg/dataflatten)
+
+package profiles
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+const profilesPath = "/usr/bin/profiles"
+
+const userAllowedCharacters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-_"
+
+type Table struct {
+	client    *osquery.ExtensionManagerClient
+	logger    log.Logger
+	tableName string
+}
+
+func TablePlugin(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+
+	columns := []table.ColumnDefinition{
+		table.TextColumn("fullkey"),
+		table.TextColumn("parent"),
+		table.TextColumn("key"),
+		table.TextColumn("value"),
+		table.TextColumn("query"),
+
+		// profiles options. See `man profiles`. These may not be needed,
+		// we use `show -all` as the default, and it probably covers
+		// everything.
+		table.TextColumn("user"),
+		table.TextColumn("command"),
+		table.TextColumn("type"),
+	}
+
+	t := &Table{
+		client:    client,
+		logger:    logger,
+		tableName: "kolide_profiles",
+	}
+
+	return table.NewPlugin(t.tableName, columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	for _, command := range tablehelpers.GetConstraints(queryContext, "command", tablehelpers.WithAllowedCharacters("abcdefghijklmnopqrstuvwxyz"), tablehelpers.WithDefaults("show")) {
+		for _, profileType := range tablehelpers.GetConstraints(queryContext, "type", tablehelpers.WithAllowedCharacters("abcdefghijklmnopqrstuvwxyz"), tablehelpers.WithDefaults("")) {
+			for _, user := range tablehelpers.GetConstraints(queryContext, "user", tablehelpers.WithAllowedCharacters(userAllowedCharacters), tablehelpers.WithDefaults("_all")) {
+				for _, dataQuery := range tablehelpers.GetConstraints(queryContext, "query", tablehelpers.WithDefaults("")) {
+
+					profileArgs := []string{command, "-output", "stdout-xml"}
+
+					if profileType != "" {
+						profileArgs = append(profileArgs, "-type", profileType)
+					}
+
+					switch {
+					case user == "" || user == "_all":
+						profileArgs = append(profileArgs, "-all")
+					case user == "_device":
+						break
```

Comment:
why the `break` here? I assume that `_device` is a special value for user?

Testing out the `/usr/bin/profiles` command, this might be an invalid value?

Reply:
Ah. this has a weird UI. I'm overloading the `users` column to allow selecting:
1. all profiles merged, using `_all` (or empty, as it's the default)
2. device profiles, using `_device` which corresponds to no additional command line flags. Thus break
3. user specific ones.

I added a comment